models/constants.rb

```diff
@@ -28,3 +28,6 @@
 EXTERNAL_ID = "external_id".freeze
 COMMENT = "comment".freeze
 THREAD = "thread".freeze
+
+RETIRED_TITLE = "<removed_text>".freeze
```

Comment:
Since you specifically asked about this, I think it's fine.  You could also go the reddit "[deleted]" route.